iOSPhotoEditor/PhotoEditor+Controls.swift

```diff
@@ -20,13 +20,21 @@ extension PhotoEditorViewController {
         case draw
         case text
         case clear
+        case reset
     }
 
     @IBAction func cancelButtonTapped(_ sender: Any) {
-        photoEditorDelegate?.canceledEditing()
-        self.dismiss(animated: true, completion: nil)
+        guard undoManager?.canUndo == true else {
+            cancel()
+            return
+        }
         
-        tracker?.track(event: .cancel)
+        let alert = UIAlertController(title: "Möchten Sie wirklich alle Markierungen verwerfen?", message: nil, preferredStyle: .actionSheet)
+        alert.addAction(UIAlertAction(title: "Änderungen verwerfen", style: .destructive) { _ in
+            self.cancel()
+        })
+        alert.addAction(UIAlertAction(title: "Weiter bearbeiten", style: .cancel))
+        present(alert, animated: true, completion: nil)
```

Comment:
Führt zum Crash auf dem iPad
_*** Terminating app due to uncaught exception 'NSGenericException', reason: 'Your application has presented a UIAlertController (<UIAlertController: 0x7f77bf052800>) of style UIAlertControllerStyleActionSheet from UINavigationController (<UINavigationController: 0x7f77df037600>). The modalPresentationStyle of a UIAlertController with this style is UIModalPresentationPopover. You must provide location information for this popover through the alert controller's popoverPresentationController. You must provide either a sourceView and sourceRect or a barButtonItem. If this information is not known when you present the alert controller, you may provide it in the UIPopoverPresentationControllerDelegate method -prepareForPopoverPresentation.'_